### PR TITLE
Add password and email validation for /users

### LINKED ISSUE #28: Add user validation for /users patch endpoint
Currently any student can patch other students' profile details. Need to update the /users PATCH endpoint to restrict students to only update their own profile, but still allow teachers to update any profile.

### PR DESCRIPTION
**Pull request by gremble0**

### CODE REVIEW

**Review by JosteinRuen — APPROVED** [review]
Superb 👍

**Review by thomaafl — APPROVED** [review]

File: src/controllers/user.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import User from '../domain/user.js'
 import bcrypt from 'bcrypt'
-import { validateTeacherRole } from '../middleware/auth.js'
```
Comment (allendewilzin) [inline src/controllers/user.js:3]: explain why this is removed please

Comment (gremble0) [reply src/controllers/user.js:3]: It was previously used in line 122 inside the updateById function, but its no longer used.

Comment (gremble0) [reply src/controllers/user.js:3]: Validation is instead done by these helper functions:

```javascript
const studentUpdatingOther = (req) => { ... }
```

```javascript 
const studentUpdatingIllegalFields = (req) => { ... }
```

**Review by allendewilzin — APPROVED** [review]

**Pull request merged**